pkg/policy/mapstate.go

```diff
@@ -292,23 +320,30 @@ func (keys MapState) deleteKeyWithChanges(key Key, owner MapStateOwner, adds, de
 // denyPreferredInsertWithChanges inserts a key and entry into the map by giving preference
 // to deny entries, and L3-only deny entries over L3-L4 allows.
 // Incremental changes performed are recorded in 'adds' and 'deletes', if not nil.
-func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStateEntry, adds, deletes Keys) {
+func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStateEntry, adds, deletes Keys, identities Identities) {
 	allCpy := allKey
 	allCpy.TrafficDirection = newKey.TrafficDirection
 	// If we have a deny "all" we don't accept any kind of map entry
 	if v, ok := keys[allCpy]; ok && v.IsDeny {
 		return
 	}
+	newEntryNets := newEntry.getNets(identities, identity.NumericIdentity(newKey.Identity))
 
 	if newEntry.IsDeny {
-		// case for an existing allow L4-only and we are inserting deny L3-only
 		switch {
+		// case for inserting a deny L3-only
 		case newKey.DestPort == 0 && newKey.Nexthdr == 0 && newKey.Identity != 0:
 			l4OnlyAllows := MapState{}
+			// look for broad allows "all/unknown" identity and insert the same
+			// entry, but with the specifc L3-deny identity. This ensures that
```

Comment:
typo

Reply:
There's still a few typos in this comment. `specifc`, `as`